test/e2e/scheduling/framework.go

```diff
@@ -51,22 +54,57 @@ func WaitForStableCluster(c clientset.Interface, masterNodes sets.String) int {
 			framework.Failf("Timed out after %v waiting for stable cluster.", timeout)
 			break
 		}
-		allScheduledPods, allNotScheduledPods = getFilteredPods(c, masterNodes, metav1.NamespaceAll)
+		allScheduledPods, allNotScheduledPods = getScheduledAndUnscheduledPods(c, masterNodes, metav1.NamespaceAll)
 	}
 	return len(allScheduledPods)
 }
 
-// getFilteredPods lists scheduled and not scheduled pods in the given namespace, with succeeded and failed pods filtered out.
-func getFilteredPods(c clientset.Interface, masterNodes sets.String, ns string) (scheduledPods, notScheduledPods []v1.Pod) {
+// WaitForPodsToBeDeleted waits until pods that are terminating to get deleted.
+func WaitForPodsToBeDeleted(c clientset.Interface) {
+	startTime := time.Now()
+	deleting := getDeletingPods(c, metav1.NamespaceAll)
+	for len(deleting) != 0 {
+		if startTime.Add(timeout).Before(time.Now()) {
+			framework.Logf("Pods still not deleted")
+			for _, p := range deleting {
+				framework.Logf("%v/%v", p.Namespace, p.Name)
+			}
+			framework.Failf("Timed out after %v waiting for pods to be deleted", timeout)
+			break
+		}
+		time.Sleep(wait)
+		deleting = getDeletingPods(c, metav1.NamespaceAll)
+	}
+}
+
+// getScheduledAndUnscheduledPods lists scheduled and not scheduled pods in the given namespace, with succeeded and failed pods filtered out.
+func getScheduledAndUnscheduledPods(c clientset.Interface, masterNodes sets.String, ns string) (scheduledPods, notScheduledPods []v1.Pod) {
 	pods, err := c.CoreV1().Pods(ns).List(metav1.ListOptions{})
 	framework.ExpectNoError(err, "listing all pods in kube-system namespace while waiting for stable cluster")
```

Comment:
update this as well, listing pods in the given namespace.

Reply:
Done